### PR TITLE
Clipdrop integration

### LINKED ISSUE #8226: [Integration]: Integrate Clipdrop with MindsDB
### Is there an existing integration?

- [X] I have searched the existing integrations.

### Use Case

The goal of this issue is to build a new handler that will utilize the Clipdrop API to connect MindsDB to Clipdrop.

## Resources 💡

🏁 To get started, please refer to the following resources:

📚 [Building Application Handler docs](https://docs.mindsdb.com/contribute/app-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

📖 [Clipdrop](https://clipdrop.co/)

👉 [Twitter Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/twitter_handler): This existing handler connects MindsDB with the Twitter API, allowing users to automate tasks related to posting and analyzing tweets. You can use the Twitter Handler as a reference for building the Slack API integration.

👉 [Binance API Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/binance_handler): Another example of a successful MindsDB integration to the Binance API to make trade data available to use for model training and predictions.

## Next Steps 🕵️‍♂️ 🕵️‍♀️

You can familiarize yourself with the [Clipdrop](https://clipdrop.co/)
Create a new handler that connects MindsDB to clipdrop API.
Implement the core features mentioned above.
Thoroughly test the integration to ensure proper functionality.
Submit a pull request with the new clipdrop Integration handler.

## 📢 Additional rewards 🏆
Each App integration brings 7️⃣ 0️⃣ : points for SWAG and entry into the draw for a 👇 :

💻 [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
👕 🐻 [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
ℹ️ For more info check out https://mindsdb.com/hacktoberfest/ 👈

### PR DESCRIPTION
**Pull request by Better-Boy**
## Description

Clipdrop integration

Fixes #8226 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        ----------
+        name : str
+            name of a handler instance
+        """
+        super().__init__(name)
+
+        connection_data = kwargs.get("connection_data", {})
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+        self.client = ClipdropClient(connection_data.get("api_key"),
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py:36]: Maybe move this to the connect method?

File: mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        Returns
+        -------
+        StatusResponse
+            Status confirmation
+        """
+        # FYI - No way currently to establish a connection with the given key
+        # without making a requests that is chargeable
+        # TODO - Find a way to solve FYI
+        self.is_connected = True
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py:68]: Isn't there a healthcheck method that we can call on the client?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Also check the failing test for the pep8 warnings

**Review by MinuraPunchihewa — COMMENTED** [review]
@ZoranPandovski Shouldn't this be implemented as a ML handler rather than a API handler?

File: mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py (hunk 1 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py]: Done

File: mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py (hunk 2 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py:89]: The docs don't mention any healthcheck method.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
@Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

**Comment by Better-Boy** [comment]
> @Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

@MinuraPunchihewa should I need to convert it to ML handler? If yes, I'll need help from you on https://mindsdbcommunity.slack.com/archives/C04EWFQEFEV/p1698317162200199

**Comment by Better-Boy** [comment]
> @Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

@MinuraPunchihewa can clipdrop integration be an API handler for now? Can the conversion to ML handler be taken in another PR?

**Comment by MinuraPunchihewa** [comment]
Hey @Better-Boy,
I'm sorry, but I don't think so. Both App and Database handlers are classified as DATA integrations. Since this integration does not really extract/generate any data as such, I don't think it is appropriate to leave this as is.
I will look at your PR and try to resolve the problem you were facing.

**Comment by Better-Boy** [comment]
> Hey @Better-Boy, I'm sorry, but I don't think so. Both App and Database handlers are classified as DATA integrations. Since this integration does not really extract/generate any data as such, I don't think it is appropriate to leave this as is. I will look at your PR and try to resolve the problem you were facing.

@MinuraPunchihewa I have converted the clipdrop handler from APIHandler to MLHandler. Please review and let me know if any changes are needed.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
@Better-Boy This looks great! If you can address a couple of things, we can go ahead and merge this,
1. Please include a few screenshots or a short video of this integration working as expected.
2. Please add some unit tests for this integration; I have outlined the process for doing so in your StabilityAI PR, but if you have any questions, please feel free to ask me here.

Thanks again for your great work!

**Comment by Better-Boy** [comment]
> @Better-Boy This looks great! If you can address a couple of things, we can go ahead and merge this,
> 
> 1. Please include a few screenshots or a short video of this integration working as expected.

I ran it on a VM and deleted the VM. But forgot to record or take screenshot. I'm sure it works. If needed, you can test it. I can provide the API key in slack DM.

> 2. Please add some unit tests for this integration; I have outlined the process for doing so in your StabilityAI PR, but if you have any questions, please feel free to ask me here.
> 
> Thanks again for your great work!

I have added the sample tests.

**Review by MinuraPunchihewa — APPROVED** [review]

**Review by dusvyat — APPROVED** [review]
lgtm

**Pull request merged**